x/inflation/types/v1beta2/params.go

```diff
@@ -0,0 +1,94 @@
+package v1beta2
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	"github.com/pkg/errors"
+)
+
+var _ paramtypes.ParamSet = (*Params)(nil)
+
+const (
+	DefaultInflationDecayFactor uint32 = 2 // years
+
+	keyInflationDecayFactor = "InflationDecayFactor"
+	keyInitialInflation     = "InitialInflation"
+	keyVariance             = "Variance"
+)
+
+func DefaultInitialInflation() sdk.Dec { return sdk.NewDec(100) }
+func DefaultVarince() sdk.Dec          { return sdk.MustNewDecFromStr("0.05") }
+
+func MaxInitialInflation() sdk.Dec { return sdk.NewDec(100) }
+func MinInitialInflation() sdk.Dec { return sdk.ZeroDec() }
+
+func MaxVariance() sdk.Dec { return sdk.NewDec(1) }
+func MinVariance() sdk.Dec { return sdk.ZeroDec() }
+
+func ParamKeyTable() paramtypes.KeyTable {
+	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair([]byte(keyInflationDecayFactor), &p.InflationDecayFactor, validateInflationDecayFactor),
+		paramtypes.NewParamSetPair([]byte(keyInitialInflation), &p.InitialInflation, validateInitialInflation),
+		paramtypes.NewParamSetPair([]byte(keyVariance), &p.Variance, validateVariance),
+	}
+}
+
+func DefaultParams() Params {
+	return Params{
+		InflationDecayFactor: DefaultInflationDecayFactor,
+		InitialInflation:     DefaultInitialInflation(),
+		Variance:             DefaultVarince(),
+	}
+}
+
+func (p Params) Validate() error {
+	if err := validateInflationDecayFactor(p.InflationDecayFactor); err != nil {
+		return err
+	}
+	if err := validateInitialInflation(p.InitialInflation); err != nil {
+		return err
+	}
+	if err := validateVariance(p.Variance); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateInflationDecayFactor(i interface{}) error {
+	v, ok := i.(uint32)
+	if !ok || v < 1 {
+		return errors.Wrapf(ErrInvalidParam, "%T", i)
+	}
+
+	return nil
+}
+
+func validateInitialInflation(i interface{}) error {
+	v, ok := i.(sdk.Dec)
```

Comment:
Will this work if the param type is a string? https://github.com/ovrclk/akash/pull/1352/files#diff-bbf326520f4fa8aa2e6f40542d4dfb257fe6bc3233f0d5b382b7021416619d3bR18

Reply:
That is a string only in proto definition. In Go code, it maps to `sdk.Dec` because of this configured option in the proto definition:
```
(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
```
This is how other modules in `cosmos-sdk` are mapping their parameters to `sdk.Dec`.